src/cmake-tools.ts

```diff
@@ -681,9 +672,9 @@ export class CMakeTools implements vscode.Disposable, api.CMakeToolsAPI {
   /**
    * The target that will be built with a regular build invocation
    */
-  public get defaultBuildTarget(): string|null { return this._stateManager.defaultBuildTarget; }
+  public get defaultBuildTarget(): string|null { return this.workspaceContext.state.defaultBuildTarget; }
```

Comment:
@ytimenkov Originally I was going to pass the two around as separate arguments, but the common case is to pass both, so I merged them into one object. Passing one or the other is possible, it's just a matter of finding those cases.

Nevertheless, you can initialize a `DirectoryContext` with arbitrary `ConfigurationReader` and `StateManager`, so you can dep-inject each independently freely.

The biggest purpose of this change is to eventually facilitate creating of independent `CMakeTools` instances, and being able to pass a `ConfigurationReader` objects that you instantiate yourself is useful, especially for testing so that you can just say `updatePartial()` in the unit test and observe the behavioral changes from the outside without even needing a `settings.json`. The hope is eventually that you won't even need a project to open and you can simply use directories with bare CMake projects and test them from there without needing to open a VSCode instance within them.

Ultimately, I hope to be able to run all the main tests without needing VSCode _at all_. See @KoeMai's backend-tests, which just run Mocha without VSCode.